src/services/useAppConstants.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+import supabaseClient from "../../utils/supabaseClient";
+
+type AppConstant = {
+    value: string,
+    label: string
+}
+
+const cache: Record<string, AppConstant[]> = {}
+
+const useAppConstants = (type: string) => {
+    const [init, setInit] = useState(false);
+    const [status, setStatus] = useState('idle');
+    const [data, setData] = useState<AppConstant[]>([]);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            setStatus('fetching');
+            if (!init) {
+                const response = await supabaseClient.from('app_constants')
+                    .select()
+                const types = new Set(response.data.map(c => c.type));
+                types.forEach(t => cache[t] = response.data.filter(c => c.type === t));
```

Comment:
I think lines 22 and 23 could be written as one loop over the data to be a little more clear of the intention 

```
const cache: new Map<string, AppConstant[]>()
...
response.data.forEach((c) => {
    if (!cache.has(c.type)) cach.set(c.type, [])
    cache.get(c.type).push(c)
}
...
const data = cache.get(type)
```

Reply:
ok